### PR TITLE
solved Grade school exercise

### PR DESCRIPTION
**Pull request by iuliagatea**

### CODE REVIEW

File: Ruby/grade-school/grade_school.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class School
+
+  def initialize
+    @school = Hash.new { |hash, key|  hash[key] = [] }
+  end
+
+  def add(student, grade)
+    school[grade].push(student)
```
Comment (paucio) [inline Ruby/grade-school/grade_school.rb]: You could sort the students every time you add a new one and you won't need to call `sort` in the other methods.

File: Ruby/grade-school/grade_school.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def students(grade_wanted)
+    school[grade_wanted].sort
+  end
+
+  def students_by_grade
+    school.sort.map do |grade, students|
```
Comment (paucio) [inline Ruby/grade-school/grade_school.rb]: I'm not sure if this is sorting by grade... try using `sorb_by` instead to be sure: 

https://apidock.com/ruby/Enumerable/sort_by

File: Ruby/grade-school/grade_school.rb (hunk 1 above)
Comment (asarbu) [inline Ruby/grade-school/grade_school.rb]: shouldn't we have some validations on the parameters given ? What if I do `Schoo.add("Anca", -1)` ? or `School.add(nil, 3)`

File: Ruby/grade-school/grade_school.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class School
+
+  def initialize
+    @school = Hash.new { |hash, key|  hash[key] = [] }
```
[the remark this answers is not in the document]
Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: I have this error if i eliminate the block: 
NoMethodError: undefined method `push' for nil:NilClass
Did you mean?  puts

[the remark this answers is not in the document]
Comment (asarbu) [reply Ruby/grade-school/grade_school.rb]: I meant you should use `new` with parameters, look [here](https://docs.ruby-lang.org/en/2.0.0/Hash.html#method-c-new)

[the remark this answers is not in the document]
Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: I don't know how to do that. I tried `Hash.new([])` but doesn't return what I need.

[the remark this answers is not in the document]
Comment (asarbu) [reply Ruby/grade-school/grade_school.rb]: I see now, I have given the documentation a better read and see why this is not possible. 

Please read again the documentation (if you need to), and explain to me why we cannot use `Hash.new([])`

[the remark this answers is not in the document]
Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: `Hash.new([])` will return an array of pushed values and this exercise requires that for any key pushed to return an array. The `Hash.new { |hash, key| hash[key] = [] }` will make a bond between the key and the value pushed

File: Ruby/grade-school/grade_school.rb (hunk 1 above)
Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: done

Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: done

File: Ruby/grade-school/grade_school.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class School
+
+  def initialize
+    @school = Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def add(student, grade)
+    raise ArgumentError, 'Please insert valid data' if grade < 1 || student.nil?
+    school[grade].push(student).sort!
```
Comment (asarbu) [inline Ruby/grade-school/grade_school.rb]: could you walk me through what's happening here (what is returned, what is sorted)? I am sure that it works, but at least at first sight it looks "ify"

Comment (iuliagatea) [reply Ruby/grade-school/grade_school.rb]: sorts the value of key grade, after it pushes a new value.

Comment (asarbu) [reply Ruby/grade-school/grade_school.rb]: oh ok, in this case why do you sort by grade again on line 19 ?

**Pull request merged**